File: core/src/main/java/mifos/org/mobilewallet/core/data/fineract/entity/mapper/CurrencyMapper.java
package mifos.org.mobilewallet.core.data.fineract.entity.mapper;

import javax.inject.Inject;
import javax.inject.Singleton;

import mifos.org.mobilewallet.core.domain.model.Currency;

/**
 * Created by naman on 17/8/17.
 */

@Singleton
public class CurrencyMapper {

    @Inject
    CurrencyMapper() {}

    public Currency transform(mifos.org.mobilewallet.core.data.fineract.entity.accounts.savings.Currency savingsCurrency) {

        Currency currency = new Currency();
        currency.setCode(savingsCurrency.getCode());
        currency.setDisplayLabel(savingsCurrency.getDisplayLabel());
        currency.setDisplaySymbol(savingsCurrency.getDisplaySymbol());

        return currency;
    }
}

File: core/data/src/main/java/org/mifos/mobilewallet/core/data/fineract/entity/mapper/CurrencyMapper.kt
package org.mifos.mobilewallet.core.data.fineract.entity.mapper

import com.mifos.mobilewallet.model.entity.accounts.savings.Currency
import javax.inject.Inject
import com.mifos.mobilewallet.model.domain.Currency as DomainCurrency

/**
 * Created by naman on 17/8/17.
 */
class CurrencyMapper @Inject internal constructor() {
    fun transform(savingsCurrency: Currency): DomainCurrency {
        val currency: DomainCurrency =
            DomainCurrency()
        currency.code = savingsCurrency.code
        currency.displayLabel = savingsCurrency.displayLabel
        currency.displaySymbol = savingsCurrency.displaySymbol
        return currency
    }
}
